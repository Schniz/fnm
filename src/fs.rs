use std::path::Path;

#[cfg(unix)]
pub fn symlink_dir<P: AsRef<Path>, U: AsRef<Path>>(from: P, to: U) -> std::io::Result<()> {
    std::os::unix::fs::symlink(from, to)?;
    Ok(())
}

#[cfg(windows)]
pub fn symlink_dir<P: AsRef<Path>, U: AsRef<Path>>(from: P, to: U) -> std::io::Result<()> {
    junction::create(from, to)?;
    Ok(())
}

#[cfg(windows)]
pub fn remove_symlink_dir<P: AsRef<Path>>(path: P) -> std::io::Result<()> {
    std::fs::remove_dir(path)?;
    Ok(())
}

#[cfg(unix)]
pub fn remove_symlink_dir<P: AsRef<Path>>(path: P) -> std::io::Result<()> {
    std::fs::remove_file(path)?;
    Ok(())
}

pub fn shallow_read_symlink<P: AsRef<Path>>(path: P) -> std::io::Result<std::path::PathBuf> {
    std::fs::read_link(path)
}
