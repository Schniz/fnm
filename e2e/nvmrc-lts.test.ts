import { script } from "./shellcode/script.js"
import { Bash, Fish, PowerShell, Zsh } from "./shellcode/shells.js"
import fs from "node:fs/promises"
import path from "node:path"
import describe from "./describe.js"
import testCwd from "./shellcode/test-cwd.js"

for (const shell of [Bash, Fish, PowerShell, Zsh]) {
  describe(shell, () => {
    test(`uses .nvmrc with lts definition`, async () => {
      await fs.writeFile(path.join(testCwd(), ".nvmrc"), `lts/dubnium`)

      await script(shell)
        .then(shell.env({}))
        .then(shell.call("fnm", ["install"]))
        .then(shell.call("fnm", ["use"]))
        .then(
          shell.scriptOutputContains(shell.call("fnm", ["ls"]), "lts-dubnium")
        )
        .takeSnapshot(shell)
        .execute(shell)
    })
  })
}
