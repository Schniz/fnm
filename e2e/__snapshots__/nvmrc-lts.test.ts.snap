// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bash uses .nvmrc with lts definition: Bash 1`] = `
"set -e
eval "$(fnm env)"
fnm install
fnm use
(fnm ls) | grep lts-dubnium || (echo "Expected output to contain lts-dubnium" && exit 1)"
`;

exports[`Fish uses .nvmrc with lts definition: Fish 1`] = `
"fnm env | source
fnm install
fnm use
begin; fnm ls; end | grep lts-dubnium; or echo "Expected output to contain lts-dubnium" && exit 1"
`;

exports[`PowerShell uses .nvmrc with lts definition: PowerShell 1`] = `
"$ErrorActionPreference = "Stop"
fnm env | Out-String | Invoke-Expression
fnm install
fnm use
$($__out__ = $(fnm ls | Select-String lts-dubnium); if ($__out__ -eq $null) { exit 1 } else { $__out__ })"
`;

exports[`Zsh uses .nvmrc with lts definition: Zsh 1`] = `
"set -e
eval "$(fnm env)"
fnm install
fnm use
(fnm ls) | grep lts-dubnium || (echo "Expected output to contain lts-dubnium" && exit 1)"
`;
